### PR TITLE
JBEHAVE-1335 Add possibility to skip base story if Given story failed

### PR DESCRIPTION
**Pull request by tshkursk**

### CODE REVIEW

File: jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java [hunk 1]
```diff
@@ -880,7 +880,9 @@ public void perform(RunContext context) throws InterruptedException {
                 }
                 context.reporter().afterBeforeStorySteps();
                 performGivenStories(context, givenStories, story.getGivenStories());
-                performScenarios(context);
+                if (!context.failureOccurred() || !context.configuration().storyControls().skipScenariosAfterFailure()) {
```
[the remark this answers is not in the document]
Comment (tshkursk) [reply jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java]: 'skipScenariosAfterGivenFailure' flag added

File: examples/core/src/main/java/org/jbehave/examples/core/stories/failing/skip_story_if_given_story_failed.story [hunk 2]
```diff
@@ -0,0 +1,5 @@
+GivenStories: org/jbehave/examples/core/stories/failing/non_successful.story
+
+Scenario: A scenario that is not executed because it followed a failed given story
+
+Given I do nothing
```
Comment (valfirst) [inline examples/core/src/main/java/org/jbehave/examples/core/stories/failing/skip_story_if_given_story_failed.story]: add new line to the end of file

File: jbehave-core/src/test/java/org/jbehave/core/embedder/PerformableTreeBehaviour.java [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
```
Comment (valfirst) [inline jbehave-core/src/test/java/org/jbehave/core/embedder/PerformableTreeBehaviour.java]: use `assertThat`

File: jbehave-spring/src/test/java/org/jbehave/core/configuration/spring/configuration.xml [hunk 4]
```diff
@@ -29,6 +29,7 @@
   <bean class="org.jbehave.core.configuration.spring.SpringStoryControls">
     <property name="dryRun" value="true" />
     <property name="skipScenariosAfterFailure" value="true" />
+    <property name="skipStoryIfGivenStoryFailed" value="true" />
```
Comment (valfirst) [inline jbehave-spring/src/test/java/org/jbehave/core/configuration/spring/configuration.xml]: it should be `false` by default for backward compatibility

**Pull request merged**